### PR TITLE
Add source docs, update GIFs, add template repo docs

### LINKED ISSUE #839: Document plugins
## Overview

Write documentation that describes how to implement a source or executor plugin.

## Acceptance Criteria

- Document plugin management for the end-user
   - State that the only the plugins from our repo are safe (recommended)
- Update configuration for "default" sources and executors
- Prepare documentation for adding new plugins to Botkube
    - It should contain how to verify plugin functionality in PR checks
    - Boilerplate template for plugin development. Please refer [here](https://github.com/huseyinbabal/botkube-plugins/tree/main/plugins/kubectl) for an example
    - Once needed, provide a mocking mechanism to write tests for specific plugin
    - You can see the filters docs: https://docs.botkube.io/usage/filters/development
    - You can use echo executor etc.
    - Rules about accepting plugins to our repo

## Reason

#844 

Related:

#925

### PR DESCRIPTION
**Pull request by mszostok**
## Description

Changes proposed in this pull request:

- Add source docs, 
- Update GIFs, 
- Add template repo docs

Preview URL: https://plugin-source.botkube-docs.pages.dev/next/plugin/custom-source

## Related issue(s)

Fix https://github.com/kubeshop/botkube/issues/839

### CODE REVIEW

**Comment by mszostok** [comment]
All links with prefix `https://github.com/mszostok/botkube-plugins-template` will be replaced with `https://github.com/kubeshop/botkube-plugins-template`, after setting up the repository under kubeshop

File: docs/plugin/local-testing.md [hunk 1]
```diff
@@ -44,31 +90,71 @@ This document describes steps for running Botkube core locally together with a l
...
+
+```bash
+# rebuild plugins only for current GOOS and GOARCH
+make build-plugins-single &&
+# regenerate index
+env PLUGIN_DOWNLOAD_URL_BASE_PATH="http://localhost:8080/dist" make gen-plugin-index &&
+# remove cached plugins
+rm -rf $BOTKUBE_PLUGINS_CACHE__DIR &&
+# start Botkube
+${BOTKUBE_REPO_PATH}/botkube
```
Comment (pkosiec) [inline docs/plugin/local-testing.md:155]: There wasn't any Botkube binary building step - I'd suggest changing it to
```suggestion
cd ${BOTKUBE_REPO_PATH} && go run cmd/botkube/main.go
```

File: docs/plugin/local-testing.md [hunk 2]
```diff
@@ -16,23 +16,69 @@ This document describes steps for running Botkube core locally together with a l
...
+         channels:
+           default:
+             name: random
+             bindings:
+               executors:
+                 - 'plugin-based'
+               sources:
+                 - 'plugin-based'
+         appToken: "" # provide your token starting with xapp-1-
+         botToken: "" # provide your token starting with xoxb-
```
Comment (pkosiec) [inline docs/plugin/local-testing.md:76]: Please also add:

```yaml
settings:
  clusterName: local-dev
```
To make the output messages look a bit better 😄

**Review by pkosiec — COMMENTED** [review]
Good work! A few small comments. See also my review on https://github.com/kubeshop/botkube-plugins-template/pull/2.

File: docs/plugin/template.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+- The [`ticker`](https://github.com/mszostok/botkube-plugins-template/blob/main/cmd/ticker/main.go) source that emits an event each time the configured time duration elapses.
+
+and two example release jobs:
+
+- [GitHub releases](https://github.com/mszostok/botkube-plugins-template/blob/main/.github/workflows/release.yml)
+- [GitHub Pages](https://github.com/mszostok/botkube-plugins-template/blob/main/.github/workflows/pages-release.yml)
+
+## Use template
+
+1. Navigate to [`botkube-plugins-template`](https://github.com/mszostok/botkube-plugins-template).
```
[the remark this answers is not in the document]
Comment (mszostok) [reply docs/plugin/template.md:19]: will do, I mentioned that here: https://github.com/kubeshop/botkube-docs/pull/197#issuecomment-1382913082

File: docs/plugin/local-testing.md [hunk 4]
```diff
@@ -16,23 +16,69 @@ This document describes steps for running Botkube core locally together with a l
...
+   EOF
+   ```
+
+   </details>
+
 3. Start static plugins' server:
 
    ```bash
```
[the remark this answers is not in the document]
Comment (mszostok) [reply docs/plugin/local-testing.md:84]: the `go run test/helpers/plugin_server.go` is a bit different and it builds and returns the index on each query, so the only thing is to only rebuild the binaries. As a result, it's a bit faster. Later IMO we should just create some CLI under `/hack` or `/cmd` that will help with all those plugin testing things and use it in all places 👍

File: docs/plugin/local-testing.md [hunk 5]
```diff
@@ -16,23 +16,69 @@ This document describes steps for running Botkube core locally together with a l
...
      # Enable a given communication platform and define bindings to a given executor and source plugins.
    ```
 
+   For example, for Slack and example `echo` and `ticker` plugins, run:
+   :::note
+   Do not forget to fill the `appToken` and `botToken` properties.
+   :::
+
+   <details>
+     <summary>Create /tmp/config-values.yaml</summary>
```
[the remark this answers is not in the document]
Comment (mszostok) [reply docs/plugin/local-testing.md:42]: Actually, I removed the `Edit the comm_config.yaml file.`. As later, I suggest to use:
```bash
export BOTKUBE_CONFIG_PATHS="${BOTKUBE_REPO_PATH}/helm/botkube/values.yaml,/tmp/config-values.yaml"
```

File: docs/plugin/custom-source.md [hunk 6]
```diff
@@ -0,0 +1,232 @@
+---
```
[the remark this answers is not in the document]
Comment (mszostok) [reply docs/plugin/custom-source.md:1]: yes, I planned to do that but one all of the above will be resolved. Probably I will just do that in a dedicated PR 👍

File: docs/plugin/custom-source.md [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+   				out.Output <- []byte("Ticker Event")
+   			}
+   		}
+   	}()
+
+   	return out, nil
+   }
+
+
+   // mergeConfigs merges all input configuration. In our case we don't have complex merge strategy,
```
Comment (pkosiec) [inline docs/plugin/custom-source.md:127]: This somehow doesn't render properly:

https://plugin-source.botkube-docs.pages.dev/next/plugin/custom-source/

obraz

PTAL and fix it. Thanks!

File: docs/plugin/local-testing.md [hunk 8]
```diff
@@ -6,34 +6,83 @@ sidebar_position: 6
...
+     repositories:
+       local-repo:
+         url: http://localhost:8080/plugins-index.yaml
+
+   executors:
+     "plugin-based":
+       local-repo/echo:
+         enabled: true
+         config:
+           changeResponseToUpperCase: true
```
Comment (pkosiec) [inline docs/plugin/local-testing.md:57]:
```suggestion
           transformResponseToUpperCase: true
```

File: docs/plugin/local-testing.md [hunk 9]
```diff
@@ -6,34 +6,83 @@ sidebar_position: 6
...
+   sources:
+     "plugin-based":
+       local-repo/ticker:
+         enabled: true
+         config:
+           duration: 5s
```
Comment (pkosiec) [inline docs/plugin/local-testing.md:63]:
```suggestion
           interval: 5s
```

File: docs/plugin/local-testing.md [hunk 10]
```diff
@@ -6,34 +6,83 @@ sidebar_position: 6
...
+       local-repo/source-name: # Plugin name syntax: <repo>/<plugin>[@<version>]. If version is not provided, the latest version from repository is used.
          enabled: true
          config: {} # Plugin's specific configuration.
 
    communications:
      # Enable a given communication platform and define bindings to a given executor and source plugins.
    ```
 
-3. Start static plugins' server:
+   For example, for Slack and example `echo` and `ticker` plugins provide `appToken` and `botToken` and create `/tmp/config-values.yaml` with the following values:
```
Comment (pkosiec) [inline docs/plugin/local-testing.md:40]: If this is actually a script, then maybe this is better:
```suggestion
   For example, for Slack and example `echo` and `ticker` plugins, provide `appToken` and `botToken` and run the script:
```

**Review by pkosiec — APPROVED** [review]
LGTM, but please see the comments before merge, and remember about plugin docs sync to 0.17 🙁

**Pull request merged**